framework/src/wifi_manager/wifi_manager.c

```diff
@@ -0,0 +1,440 @@
+			wifi_cb->softap_sta_leave();
+		} else {
+			ndbg("Callback wifimanager ap_disconnected failed\n");
+		}
+	}
+#ifdef CONFIG_ENABLE_IOTIVITY
+	__tizenrt_manual_linkset("del");
+#endif
+
+	/* TODO: Import files from source
+	 * sendStatusTrigger(TRIGGER_NETWORK_CHANGED, DAWIT_NW_CHANGED_DOWN);
+	 */
+}
+
+/**
+ * Public API
+ */
+
+wifi_manager_result_e wifi_manager_connect_ap(wifi_manager_ap_config_s *config)
+{
+	if (config == NULL) {
+		return WIFI_MANAGER_INVALID_ARGS;
+	}
+
+	wifi_utils_info info;
+	wifi_utils_ap_connect_config util_config;
+
+	wifi_utils_get_info(&info);
+
+	if (info.wifi_status == WIFI_UTILS_SOFT_AP_MODE) {
+		ndbg("Current mode soft ap mode, can not connect ap");
+		return WIFI_MANAGER_FAIL;
+	}
+
+	wifi_mutex_acquire(&w_mutex, WIFI_UTILS_FOREVER);
+
+	strncpy(util_config.ssid, config->ssid, config->ssid_length);
+	util_config.ssid_length = config->ssid_length;
+	strncpy(util_config.passphrase, config->passphrase, config->passphrase_length);
+	util_config.passphrase_length = config->passphrase_length;
+	util_config.ap_auth_type = config->ap_auth_type;
+	util_config.ap_crypto_type = config->ap_crypto_type;
+
+	wifi_utils_result result = wifi_utils_connect_ap(&util_config);
+	if (result != WIFI_UTILS_SUCCESS) {
+		ndbg("Wifi AP connect fail");
+		wifi_mutex_release(&w_mutex);
+		return WIFI_MANAGER_FAIL;
+	}
+
+	/* g_manager_info.wmcb->sta_connected(); */
+	wifi_utils_get_info(&info);
+
+	char ip4_add_str[18] = { 0, };
+	wifi_net_ip4_addr_to_ip4_str(info.ip4_address, ip4_add_str);
+	strncpy(g_manager_info.ssid, config->ssid, 32);
+	strcpy(g_manager_info.ip4_address, ip4_add_str);
+	g_manager_info.rssi = info.rssi;
+
+	wifi_mutex_release(&w_mutex);
+
+	return WIFI_MANAGER_SUCCESS;
+}
+
+wifi_manager_result_e wifi_manager_disconnect_ap(void)
+{
+	wifi_utils_result result = WIFI_UTILS_SUCCESS;
+
+	wifi_mutex_acquire(&w_mutex, WIFI_UTILS_FOREVER);
+	result = wifi_utils_disconnect_ap();
+	wifi_mutex_release(&w_mutex);
+
+	return result;
+}
+
+wifi_manager_result_e wifi_manager_init(wifi_manager_cb_s *wmcb)
+{
+	wifi_utils_result result = WIFI_UTILS_SUCCESS;
+
+	result = wifi_mutex_create(&w_mutex);
+	if (result != WIFI_UTILS_SUCCESS) {
+		ndbg("wifi_mutex_create fail");
+		return WIFI_MANAGER_FAIL;
+	}
+
+	result = wifi_utils_init();
+
+	if (result != WIFI_UTILS_SUCCESS) {
+		ndbg("wifi_utils_init fail");
+		return WIFI_MANAGER_FAIL;
+	}
+
+	wifi_utils_register_connection_callback(ap_connect_event_func, ap_disconnect_event_func);
+	strncpy(g_manager_info.ip4_address, "", 18);
+	strcpy(g_manager_info.ssid, "");
+	g_manager_info.rssi = 0;
+	g_manager_info.mode = STA_MODE;
+	g_manager_info.mode = AP_DISCONNECTED;
+	g_manager_info.wmcb = wmcb;
+
+#ifdef CONFIG_ENABLE_IOTIVITY
+	struct mq_attr lq_attr;
+	lq_attr.mq_maxmsg = 10;
+	lq_attr.mq_msgsize = 4;
+	lq_attr.mq_flags = 0;
+	g_dw_nwevent_mqfd = mq_open("netlink_evtq", O_WRONLY | O_CREAT, 0666, &lq_attr);
+
+	if (g_dw_nwevent_mqfd == (mqd_t) - 1) {
```

Comment:
```
(mqd_t) - 1 ==> (mqd_t)-1
```

Reply:
-1 is replaced with ERROR.
(mqd_t) - 1 ==> (mqd_t)ERROR